"""
Device details should be given exception
"""
from django.test import override_settings

from ib_users.utils.custom_test_utils import CustomTestUtils

from . import APP_NAME, OPERATION_NAME, REQUEST_METHOD, URL_SUFFIX

REQUEST_BODY = """
{
    "google_access_token": "social_id_2"
}
"""

TEST_CASE = {
    "request": {
        "path_params": {},
        "query_params": {},
        "header_params": {},
        "securities": {},
        "body": REQUEST_BODY,
    },
}


class TestCase02LoginWithGoogleV1APITestCase(CustomTestUtils):
    app_name = APP_NAME
    operation_name = OPERATION_NAME
    request_method = REQUEST_METHOD
    url_suffix = URL_SUFFIX
    test_case_dict = TEST_CASE

    def setUp(self):
        user_account_objs = self.populate_user_account()
        self.foo_user = user_account_objs[0]
        self.populate_user_social_account()

    @override_settings(ENABLE_USER_DEVICES_INFORMATION="TRUE")
    def test_case(self):
        self.default_test_case() # Returns response object.
        # Which can be used for further response object checks.
        # Add database state checks here.
